Core/src/org/sleuthkit/autopsy/coreutils/AppDBParserHelper.java

```diff
@@ -0,0 +1,1145 @@
+            }
+            if (!StringUtils.isEmpty(threadId)) {
+                msgArtifact.addAttribute(new BlackboardAttribute(ATTRIBUTE_TYPE.TSK_THREAD_ID, moduleName, threadId));
+            }
+
+
+            // Add other specified attributes
+            for (BlackboardAttribute otherAttribute: otherAttributesList) {
+                msgArtifact.addAttribute(otherAttribute);
+            }
+
+            // Find/create an account instance for sender
+            if (fromAddress != null) {
+                AccountFileInstance senderAccountInstance = createAccountInstance(accountsType, fromAddress.getUniqueID());
+
+               // Create a relationship between selfAccount and sender account
+               if (selfAccountInstance != null) {
+                   addRelationship (selfAccountInstance, senderAccountInstance, msgArtifact, Relationship.Type.MESSAGE, dateTime );
+               }
+            }
+
+            // Find/create an account instance for each recipient  
+            if (recipientsList != null) {
+                for(Account.Address recipient : recipientsList) {
+
+                   AccountFileInstance recipientAccountInstance = createAccountInstance(accountsType, recipient.getUniqueID());
+
+                   // Create a relationship between selfAccount and recipient account
+                   if (selfAccountInstance != null) {
+                       addRelationship (selfAccountInstance, recipientAccountInstance, msgArtifact, Relationship.Type.MESSAGE, dateTime );
+                   }
+                }
+            }
+
+            // post artifact 
+            Case.getCurrentCase().getSleuthkitCase().getBlackboard().postArtifact(msgArtifact, this.moduleName);
+        
+        
+        } catch (TskCoreException ex) {
+            logger.log(Level.SEVERE, "Unable to add message artifact", ex); //NON-NLS
+            return null;
+        } 
+        catch (Blackboard.BlackboardException ex) {
+            logger.log(Level.SEVERE, String.format("Unable to post artifact %s", ((msgArtifact != null)? msgArtifact.getArtifactID() : "")), ex);  //NON-NLS
+        }
+        
+        // return the artifact
+        return msgArtifact; 
+    }
+     
+    /**
+     * Adds a TSK_CALLLOG artifact. 
+     * 
+     * Also creates an account instance for the caller/receiver, and creates a 
+     * relationship between the self account and the caller/receiver account.
+     * 
+     * @param otherAccountUniqueID unique id for the caller/receiver account
+     * @param direction call direction
+     * @param fromPhoneNumber originating phone number, may be empty
+     * @param toPhoneNumber recipient phone number, may be empty
+     * @param startDateTime start date/time
+     * @param endDateTime end date/time
+     * @param contactName contact name, may be empty
+     * 
+     * @return call log artifact
+     */
+    public BlackboardArtifact addCalllog( String otherAccountUniqueID, 
```

Comment:
Not to complicate things, but would be not be shooting ourselves in the foot if we wanted to support group calls? Should this have a similar API to messages?